From 20bc5eb24ac17b4906ffb519fb64e4a19cc5be79 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 4 May 2018 12:43:49 -0700
Subject: [PATCH 2/2] mips64: Disable attribute-alias warning

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 arch/mips/kernel/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/mips/kernel/Makefile b/arch/mips/kernel/Makefile
index eb92e52eb3db..795aa4c9ef7f 100644
--- a/arch/mips/kernel/Makefile
+++ b/arch/mips/kernel/Makefile
@@ -4,6 +4,8 @@
 #
 CFLAGS_signal.o		+= $(call cc-disable-warning, attribute-alias)
 CFLAGS_syscall.o	+= $(call cc-disable-warning, attribute-alias)
+CFLAGS_linux32.o	+= $(call cc-disable-warning, attribute-alias)
+CFLAGS_signal32.o	+= $(call cc-disable-warning, attribute-alias)
 
 extra-y		:= head.o vmlinux.lds
 
-- 
2.5.0

