From 6c361abfc4a47a67f2f40a2ce0b92c11ae5fa4ef Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Wed, 23 Jan 2019 14:50:26 +0000
Subject: [PATCH] yaffs2: adjust to new location of MS_RDONLY define

Signed-off-by: OpenEmbedded <oe.patch@oe>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 fs/yaffs2/yaffs_vfs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/yaffs2/yaffs_vfs.c b/fs/yaffs2/yaffs_vfs.c
index b029b8d4276e..61ac4e4c45d1 100644
--- a/fs/yaffs2/yaffs_vfs.c
+++ b/fs/yaffs2/yaffs_vfs.c
@@ -169,6 +169,8 @@ static uint32_t YCALCBLOCKS(uint64_t partition_size, uint32_t block_size)
 #include <linux/mtd/mtd.h>
 #include <linux/cred.h>
 
+#include <uapi/linux/mount.h>
+
 #include "yportenv.h"
 #include "yaffs_trace.h"
 #include "yaffs_guts.h"
-- 
2.19.1

