From 60da2a8107e2875f2a29f907ece369506bb877bf Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Fri, 8 Jan 2016 17:51:34 +0200
Subject: [PATCH 112/278] drivers/power: Cleanup Warnings in Axxia Reset Code

Signed-off-by: John Jacques <john.jacques@intel.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 drivers/power/reset/axxia-reset.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/reset/axxia-reset.c b/drivers/power/reset/axxia-reset.c
index dd5e50f..b23a6dc 100644
--- a/drivers/power/reset/axxia-reset.c
+++ b/drivers/power/reset/axxia-reset.c
@@ -50,7 +50,7 @@
 	syscon = ioremap(SYSCON_PHYS_ADDR, SZ_64K);
 
 	if (WARN_ON(!syscon))
-		return -ENODEV;
+		return;
 
 	/* set retention reset bit in pscratch */
 	writel(0x00000001, syscon + 0xdc); /* Access Key */
-- 
1.8.5.2.233.g932f7e4

