From 8140ef093e04e8b049da3f6c3e047419d1c2c0da Mon Sep 17 00:00:00 2001
From: SangeethaRao <sangeetha.rao@intel.com>
Date: Mon, 21 Sep 2015 14:14:17 -0500
Subject: [PATCH 095/278] pci: host: Updated X9 PCIe driver to clear legacy
 interrupts

Signed-off-by: SangeethaRao <sangeetha.rao@intel.com>
---
 drivers/pci/host/pcie-axxia.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/pci/host/pcie-axxia.c b/drivers/pci/host/pcie-axxia.c
index 4268311..28d4ccc 100644
--- a/drivers/pci/host/pcie-axxia.c
+++ b/drivers/pci/host/pcie-axxia.c
@@ -599,6 +599,9 @@ static irqreturn_t axxia_pcie_irq_handler(int irq, void *arg)
 		pr_info("RADM_INTB_ASSERTED\n");
 	if (val & RADM_INTA_ASSERTED)
 		pr_info("RADM_INTA_ASSERTED\n");
+	/* Clear the legacy interrupts */
+	axxia_cc_gpreg_writel(pp, val,
+				CC_GPREG_EDG_IRQ_STAT);
 
 	if (IS_ENABLED(CONFIG_PCI_MSI)) {
 		axxia_cc_gpreg_readl(pp, CC_GPREG_EDG_IRQ_STAT_HI, &val);
-- 
1.8.5.2.233.g932f7e4

