From 91e61b10fb2cdb55596a297617e44dfe4bed48de Mon Sep 17 00:00:00 2001
From: Yue Tao <Yue.Tao@windriver.com>
Date: Tue, 28 May 2019 11:18:27 +0800
Subject: [PATCH] drivers: pinctrl: fix kernel warning unused variable 'gpr'

Signed-off-by: Yue Tao <Yue.Tao@windriver.com>
---
 drivers/pinctrl/freescale/pinctrl-imx.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/pinctrl/freescale/pinctrl-imx.c b/drivers/pinctrl/freescale/pinctrl-imx.c
index 96cef23cce64..9617d956db7f 100644
--- a/drivers/pinctrl/freescale/pinctrl-imx.c
+++ b/drivers/pinctrl/freescale/pinctrl-imx.c
@@ -576,7 +576,6 @@ int imx_pinctrl_probe(struct platform_device *pdev,
 	struct device_node *np;
 	struct imx_pinctrl *ipctl;
 	struct resource *res;
-	struct regmap *gpr;
 	int ret, i;
 
 	if (!info || !info->pins || !info->npins) {
-- 
2.17.1

