From ed2ea3dffd7f71208edcc34a7332fa1757f5f4e6 Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@lsi.com>
Date: Mon, 4 Nov 2013 14:07:15 -0600
Subject: [PATCH 072/158] drivers/tty: AXM5516 emulation bringup - amba-pl011.c

Signed-off-by: John Jacques <john.jacques@lsi.com>
---
 drivers/tty/serial/amba-pl011.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/tty/serial/amba-pl011.c b/drivers/tty/serial/amba-pl011.c
index e2774f9..6773740 100644
--- a/drivers/tty/serial/amba-pl011.c
+++ b/drivers/tty/serial/amba-pl011.c
@@ -1801,8 +1801,8 @@ pl011_set_termios(struct uart_port *port, struct ktermios *termios,
 			quot -= 2;
 	}
 	/* Set baud rate */
-	writew(quot & 0x3f, port->membase + UART011_FBRD);
-	writew(quot >> 6, port->membase + UART011_IBRD);
+	writew(0x5, port->membase + UART011_FBRD);
+	writew(0x1, port->membase + UART011_IBRD);
 
 	/*
 	 * ----------v----------v----------v----------v-----
-- 
1.8.1.2

