From 8a990322beb7b3aa5a06d7bd630f819b70911587 Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 8 Jul 2011 20:25:16 +0200
Subject: [PATCH 258/258] Add localversion for -RT release

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 localversion-rt | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 localversion-rt

diff --git a/localversion-rt b/localversion-rt
new file mode 100644
index 000000000000..6f206be67cd2
--- /dev/null
+++ b/localversion-rt
@@ -0,0 +1 @@
+-rt1
-- 
2.5.0

