From 414721c316a685c2660cb4af43866cd05dcbd08e Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Fri, 17 Nov 2017 12:46:05 -0500
Subject: [PATCH 28/69] Linux 4.9.50-rt40

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 5498386d0d0c..2af6c89aee6d 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt39
+-rt40
-- 
2.5.0

