From 50f205d9f1ddf1ee29717fbe8a59c45bd6fa0981 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Thu, 30 Nov 2017 19:12:26 -0500
Subject: [PATCH 65/69] Linux 4.9.65-rt57

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index fdb0f880c7e9..c06cc4356292 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt56
+-rt57
-- 
2.5.0

