From 0bf5f3a04255a1b962f19d9954351a8d95d926e6 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Fri, 17 Nov 2017 14:23:35 -0500
Subject: [PATCH 34/69] Linux 4.9.56-rt46

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 38c40b21a885..272158183778 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt45
+-rt46
-- 
2.5.0

