From 2124493c3fd0e75a8bbecb3207dc75f886586a93 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Wed, 2 Aug 2017 14:09:36 -0400
Subject: [PATCH 18/69] Linux 4.9.40-rt30

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 90290c642ed5..b72862e06be4 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt29
+-rt30
-- 
2.5.0

