From 074ee7c6f2bfba3c022ac0c84d9dd0b772e03b69 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Fri, 17 Nov 2017 14:25:46 -0500
Subject: [PATCH 36/69] Linux 4.9.58-rt48

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 8a777ac42aab..24707986c321 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt47
+-rt48
-- 
2.5.0

