From 1d6d96e8b3e3b09ba5ae8309d9b776816f035d4b Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Tue, 1 Aug 2017 14:47:19 -0400
Subject: [PATCH 14/69] Linux 4.9.36-rt26

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c5b71f9a229d..2e9afd4a0afd 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt25
+-rt26
-- 
2.5.0

