From a00bc26b1d4fa49b71dc8f9de046534d4a330647 Mon Sep 17 00:00:00 2001
From: Julia Cartwright <julia@ni.com>
Date: Mon, 15 Jan 2018 13:10:48 -0600
Subject: [PATCH 69/69] Linux 4.9.76-rt61

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 66fa05e70f29..9b7de9345ef4 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt60
+-rt61
-- 
2.5.0

