From 03c5a33e775b6b2c7ea3e33b45ac67ccbae2a312 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Wed, 29 Nov 2017 13:59:14 -0500
Subject: [PATCH 48/69] Linux 4.9.63-rt54

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index e6c0dc6a54cd..3165a8781ff5 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt53
+-rt54
-- 
2.5.0

