From ddedf3d3d2b5a12251ec911e4d102b17311db011 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Tue, 5 Sep 2017 11:11:55 -0400
Subject: [PATCH 24/69] Linux 4.9.46-rt36

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 366440d74b77..22940344b6c0 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt35
+-rt36
-- 
2.5.0

