From 805d1b6354c4c8c2543e39147155774a3077f536 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Fri, 17 Nov 2017 12:45:39 -0500
Subject: [PATCH 27/69] Linux 4.9.49-rt39

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 49bae8d6aa67..5498386d0d0c 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt38
+-rt39
-- 
2.5.0

