From c875b428b346f3909b0bfd6383d7938ef5e7d57e Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Thu, 14 Dec 2017 23:01:31 -0500
Subject: [PATCH 68/69] Linux 4.9.68-rt60

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 8cdeb45db6e2..66fa05e70f29 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt59
+-rt60
-- 
2.5.0

