From 5db7939cb07079bd1684759b620dede642ef4ddc Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Fri, 17 Nov 2017 14:24:06 -0500
Subject: [PATCH 35/69] Linux 4.9.57-rt47

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 272158183778..8a777ac42aab 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt46
+-rt47
-- 
2.5.0

