From 964f42799eb60dd5e529be6d1fab5aac4420292d Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Tue, 1 Aug 2017 14:48:56 -0400
Subject: [PATCH 16/69] Linux 4.9.38-rt28

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index be1e37b64ad0..5a64c9c2ab61 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt27
+-rt28
-- 
2.5.0

