From 80dcd2991e415bf7405ae5396bf72f133ca724f1 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Fri, 17 Nov 2017 16:07:37 -0500
Subject: [PATCH 39/69] Linux 4.9.61-rt51

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 42c384668389..75493460c41f 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt50
+-rt51
-- 
2.5.0

