From 6b9815ea1f5c635c2abe948ecd133e490c7e1208 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Wed, 29 Nov 2017 13:59:42 -0500
Subject: [PATCH 49/69] Linux 4.9.64-rt55

---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 3165a8781ff5..51b05e9abe6f 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt54
+-rt55
-- 
2.5.0

