From 1651c26b056443f2bc27f44f4f1076ce69730d54 Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 8 Jul 2011 20:25:16 +0200
Subject: [PATCH] Add localversion for -RT release

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>

diff --git a/localversion-rt b/localversion-rt
new file mode 100644
index 000000000000..aaf12c5b6634
--- /dev/null
+++ b/localversion-rt
@@ -0,0 +1 @@
+-rt0
-- 
2.1.4

