From 644ad424d8e08100ebccb3d03995119b246c9fb1 Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Wed, 7 Jun 2017 10:50:43 +0200
Subject: [PATCH 07/69] v4.9.30-rt21

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index e095ab819714..6c6cde1c29e3 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt20
+-rt21
-- 
2.5.0

