From 9c1e84c9b81b6bf1df55f26f2e0517266c37f7eb Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Wed, 29 May 2019 21:23:39 +0200
Subject: [PATCH 9/9] v5.0.19-rt11

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index d79dde624aaa..05c35cb58077 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt10
+-rt11
-- 
2.19.1

